### ISSUE TITLE
cli input from stdin will fail in windows environment

### ISSUE THREAD

**Issue by duaraghav8** [duaraghav8/Solium#75]
https://github.com/duaraghav8/Solium/blob/d6bba6465b6c12fc07efd325c9df29f1e78f3cf9/lib/cli.js#L178

input is currently being read using ```fs``` module from ```/dev/stdin``` - wat!? Okay so the named pipe will obviously be available in Linux environments, but not in windows. Need to account for environment in which sodium is running.

**Comment by federicobond** [comment]
`process.stdin` should work in all environments.

**Comment by duaraghav8** [comment]
So I dug deeper into this. The aim here is to read the source code from ```stdin``` **synchronously**.
This is done perfectly as of now (```var sourceCode = fs.readFileSync ('/dev/stdin', 'utf-8');```), but it won't work for windows.
[This piece](https://nodejs.org/api/process.html#process_process_stdin) is straightforward and keeps things clean, but is asynchronous. So we'd have to introduce callbacks/promises  at a lot of places.

[This](https://github.com/duaraghav8/Solium/blob/support-input-from-stdin-on-windows/lib/cli.js#L165) is a WIP.

**Comment by federicobond** [comment]
Yes, get used to using callbacks and promises a lot. `node` is not meant for synchronous stuff.

One alternative (once you configure Babel for ES6) is to use `async`/`await`.

**Comment by fvictorio** [comment]
This is how ESLint does it:

https://github.com/eslint/eslint/blob/6791d189da84bd1a06b326e645f021e4a74bd2e0/bin/eslint.js#L59-L75

It's styll async, but they do it before calling the main `execute` method. Maybe something like that could be done in `bin/solium.js`?
